### PR TITLE
Mention a chance of rain whenever an umbrella is suggested

### PR DESCRIPTION
**Pull request by mikelward**
## What & why

Aligns the prose's "chance of rain" threshold with the umbrella and the conditions strip. `PrecipProbability.POSSIBLE_THRESHOLD` drops **30% → 20%**, so all three rain surfaces now fire on the same days:

| Surface | Rain gate |
|---|---|
| Umbrella default rule | ≥ 20% chance OR a drizzle code |
| Conditions strip droplet | ≥ 20% chance OR a rain-shaped code |
| Prose "chance of rain" (POSSIBLE) | ≥ 20% (was 30%) |

Before this, a 20–29% chance would fire the umbrella and show the strip droplet while the forecast *text* stayed silent — surfaces disagreeing on the same screen.

## Notes

- `POSSIBLE_THRESHOLD` is shared, so the **week-ahead headline** tracks the same 20% bar (it's documented as only mentioning rain the today/tonight insight would).
- The **LIKELY** tier (confident "rain", not "chance of rain") is unchanged at **50%**.
- Updated the now-stale "30%" references in the prose KDocs, the `strings.xml` translator comments, and the `AGENTS.md` domain conventions.

## Test plan

- Updated two `RenderInsightSummaryTest` boundary cases that pinned the old 30% bar (now exercise the 20% bar with sub-20% per-model values).
- `./gradlew :core:domain:test :core:data:test :app:testDebugUnitTest` green; **no snapshot changes** (no preview falls in the 20–29% range).

### Follow-up still open
This closes the *threshold* half of `TODO(rain-confidence-reconcile)`. The remaining half — the rules/strip being **binary** while the prose **grades** confidence (single-model "chance" vs majority "rain") — is still open and under discussion.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01QqmafbvGC8Ld9Yo9Sv1cuh

---
_Generated by [Claude Code](https://claude.ai/code/session_01QqmafbvGC8Ld9Yo9Sv1cuh)_

### CODE REVIEW

File: core/domain/src/main/kotlin/app/clothescast/core/domain/model/Precipitation.kt [hunk 1]
```diff
@@ -185,13 +185,19 @@ internal fun PerModelHourly.peakRainCondition(): WeatherCondition =
...
+ *
+ * [POSSIBLE_THRESHOLD] is deliberately aligned with the umbrella default's
+ * probability gate ([ClothesRule.DEFAULTS]) and the conditions strip's rain-cell
+ * gate ([app.clothescast.ui.garment] `RAIN_PEAK_THRESHOLD_PCT`), so the prose
+ * mentions rain on exactly the days the umbrella fires and the strip shows a
+ * droplet — no surface flags rain the others stay silent on.
  */
 object PrecipProbability {
-    const val POSSIBLE_THRESHOLD: Double = 30.0
+    const val POSSIBLE_THRESHOLD: Double = 20.0
```
[the remark this answers is not in the document]
Comment (mikelward) [reply core/domain/src/main/kotlin/app/clothescast/core/domain/model/Precipitation.kt:200]: Good catch — fixed in `1925c9b`. The base / no-per-model fallback in `peakPrecip` no longer hardcodes `LIKELY`; it now grades by probability like the per-model tiers — `LIKELY` only at/above `LIKELY_THRESHOLD` (≥ 50%), `POSSIBLE` in the 20–49% band. So a base-only rain code at 20–29% renders as a hedged "Chance of rain", not a definite "Rain". This also corrects the pre-existing case where a 30–49% base-only forecast overstated as "Rain". Added two tests (sub-50% → POSSIBLE, ≥50% → LIKELY); full suite green, no snapshot changes.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (mikelward) [reply core/domain/src/main/kotlin/app/clothescast/core/domain/model/Precipitation.kt:200]: Good catch — fixed in `09e219c`. The POSSIBLE probability tier now keeps the driving model's own precip type: it uses the peak reading's `effectivePrecipCondition()` (drizzle / rain / snow), falling back to `perModelConditionAt` only when the reading has no code/amount. So a 20–49% per-model drizzle or snow reading over a dry blended hour reads "chance of drizzle / snow" instead of defaulting to rain. (The all-low-POP coded-drizzle case was already correct via the coded fallback; this covers the 20–29% band that now hits the probability tier first.) Added a test asserting a 25% drizzle reading keeps DRIZZLE.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (mikelward) [reply core/domain/src/main/kotlin/app/clothescast/core/domain/model/Precipitation.kt:200]: Acknowledged — real 1pp boundary mismatch (`≥ 20` for prose/strip vs strict `> 20` for the `PrecipitationProbabilityAbove` rule). Confirming the alignment direction with the maintainer before changing it; will update here once decided. Note it only diverges at *exactly* 20.0%, which is rare after the per-model blend averages probabilities.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (mikelward) [reply core/domain/src/main/kotlin/app/clothescast/core/domain/model/Precipitation.kt:200]: Fixed in `650b3581` (maintainer chose "fire at exactly 20% too"). `PrecipitationProbabilityAbove.matches` now compares inclusively (`>=`), so a forecast sitting exactly on the gate fires the umbrella the same moment the prose says "chance of rain" and the strip shows a droplet — the boundary is consistent across all three. This applies to every probability rule ("above N%" now means "at or above N%"), which is the intended behavior. Added a boundary test (exactly 50% → fires).

---
_Generated by [Claude Code](https://claude.ai/code)_
